s3/src/main/scala/akka/stream/alpakka/s3/S3Settings.scala

```diff
@@ -17,15 +17,17 @@ final case class S3Settings(bufferType: BufferType,
                             proxy: Option[Proxy],
                             credentialsProvider: AWSCredentialsProvider,
                             s3Region: String,
-                            pathStyleAccess: Boolean) {
+                            pathStyleAccess: Boolean,
+                            baseUrl: Option[String]) {
```

Comment:
In this impl this acts more like an "override s3 base url", which could be fine.
An alternative is a bit more complex, and would be alternative mappings for the s3Region strings -- and we'd by default put the global known s3 urls in there.

Reply:
I think it's probably best to go with the approach in place now. Minio, for instance, expects the region to be set to us-east-1 while using their custom endpoint.